### DISCUSSION TITLE
Bundling just CSR into a single deployable to send to clients?

### DISCUSSION THREAD

**Discussion by yatesco** [leptos-rs/leptos#3234] [Q&A] [answered]
(I'm *not* talking about using Leptos's server side capability!)

Hi - I have a client-side only app and `trunk build --release` correctly builds the `dist/index.html` and associated assets. The challenge is that I don't want to actually _host_ this anywhere, I want users to run this locally themselves and unfortunately, it's no longer possible to simply "unzip and open index.html" (because cross origin issues). And yes, I am that old ;-).

I _could_ create an Axum crate and then build that, but this seems like such an obvious use-case that I'm surprised there isn't any prior art, something like `trunk build --self-hosted` or `cargo leptos --csr --self-hosted` which would do this for me?

I realise the conventional answer is "host it somewhere dummy" but is there a better option than copying `dist/*` into my own Axum crate?

Thanks!

**Comment by yatesco** [comment]
so https://github.com/svenstaro/miniserve or https://static-web-server.net might be the best contenders...

**Reply by benwis** [reply, marked as answer]
I can't say we've considered bundling the CSR output with a server. I suspect you could write a small Rust server, and then embed all the other files for Leptos CSR into that binary to be served with it.

**Comment by psobolik** [comment]
You might consider [Tauri](https://tauri.app/), which is kind of like Electron for Rust in that it allows a web client to run as a desktop app. For a learning project, I used pnpm to generate an app with `pnpm create tauri-app`, then chose Rust for the frontend language and Leptos for the "UI template." It can even generate installers for Windows, MacOS and Linux.
